apps/backend/src/modules/activity-logs/activity-log.service.ts

```diff
@@ -0,0 +1,109 @@
+import { type Service } from "~/libs/types/types.js";
+import { type ContributorService } from "~/modules/contributors/contributors.js";
+import { type GitEmailService } from "~/modules/git-emails/git-emails.js";
+import { type ProjectApiKeyService } from "~/modules/project-api-keys/project-api-keys.js";
+
+import { ActivityLogEntity } from "./activity-log.entity.js";
+import { type ActivityLogRepository } from "./activity-log.repository.js";
+import {
+	type ActivityLogCreateRequestDto,
+	type ActivityLogGetAllResponseDto,
+} from "./libs/types/types.js";
+
+type Constructor = {
+	activityLogRepository: ActivityLogRepository;
+	contributorService: ContributorService;
+	gitEmailService: GitEmailService;
+	projectApiKeyService: ProjectApiKeyService;
+};
+
+class ActivityLogService implements Service {
+	private activityLogRepository: ActivityLogRepository;
+	private contributorService: ContributorService;
+	private gitEmailService: GitEmailService;
+	private projectApiKeyService: ProjectApiKeyService;
+
+	public constructor({
+		activityLogRepository,
+		contributorService,
+		gitEmailService,
+		projectApiKeyService,
+	}: Constructor) {
+		this.activityLogRepository = activityLogRepository;
+		this.contributorService = contributorService;
+		this.gitEmailService = gitEmailService;
+		this.projectApiKeyService = projectApiKeyService;
+	}
+
+	public async create(
+		payload: { apiKey: string } & ActivityLogCreateRequestDto,
+	): Promise<ActivityLogGetAllResponseDto> {
+		const { apiKey, items, userId } = payload;
+
+		const existingProjectApiKey =
+			await this.projectApiKeyService.findByApiKey(apiKey);
+
+		const { projectId } = existingProjectApiKey;
+
+		const createdActivityLogs: ActivityLogGetAllResponseDto = {
+			items: [],
+		};
+
+		for (const record of items) {
+			const { date, items } = record;
+
+			for (const logItem of items) {
```

Comment:
Let's create a separate helper method for this, to reduce the complexity. Something like createActivityLog.